src/ILCompiler.Compiler/src/Compiler/VirtualMethodCallHelper.cs

```diff
@@ -42,5 +42,53 @@ public static int GetVirtualMethodSlot(NodeFactory factory, MethodDesc method)
             
             return methodSlot == -1 ? -1 : baseSlots + methodSlot;
         }
+
+        /// <summary>
+        /// VTables by default are built on demand containing only the virtual methods that are reachable from
+        /// an entrypoint. To support cross-module linking BuildFullVTableForNode deterministically builds a
+        /// VTable for a type so they will be identical in app and library modules.
+        /// </summary>
+        public static void ConstructVTableForType(NodeFactory factory, TypeDesc type)
```

Comment:
This method appears to go against the declarative nature of the dependency graph. I think we need a new node type that describes vtable dependencies of external base types.

EETypeNode should just depend on that new node if the base type is not part of the compilation group.